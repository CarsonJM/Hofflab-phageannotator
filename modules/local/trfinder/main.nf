process TRFINDER {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/biopython:1.81':
        'biocontainers/biopython:1.81' }"

    input:
    tuple val(meta) , path(fasta)

    output:
    tuple val(meta), path("${prefix}_trfinder.tsv")         , emit: tr_stats
    tuple val(meta), path("${prefix}_trfinder.fasta.gz")    , emit: tr_fasta
    path "versions.yml"                                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    trfinder.py \\
        --input ${fasta} \\
        --prefix ${prefix} \\
        ${args}

    gzip -f ${prefix}_trfinder.fasta

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_trfinder.tsv
    echo "" | gzip > ${prefix}_trfinder.fasta.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
    END_VERSIONS
    """
}
