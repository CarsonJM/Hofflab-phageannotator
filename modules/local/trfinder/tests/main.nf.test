nextflow_process {

    name "Test Process: TRFINDER"
    script "../main.nf"
    process "TRFINDER"


    test("fasta.gz") {
        when {
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }


    test("fasta.gz - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [id: 'test'],
                    file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
