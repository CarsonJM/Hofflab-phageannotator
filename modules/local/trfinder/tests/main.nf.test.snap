{
    "fasta.gz - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_trfinder.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_trfinder.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "2": [
                    "versions.yml:md5,8965d3d0e4b22a572f64f1007a2439fd"
                ],
                "tr_fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "test_trfinder.fasta.gz:md5,68b329da9893e34099c7d8ad5cb9c940"
                    ]
                ],
                "tr_stats": [
                    [
                        {
                            "id": "test"
                        },
                        "test_trfinder.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8965d3d0e4b22a572f64f1007a2439fd"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-24T08:30:31.069746307"
    },
    "fasta.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_trfinder.tsv:md5,e5bebaac45561e6a7a2a3d88baa42985"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test_trfinder.fasta.gz:md5,4404f9d64e3cbdced310646cbc2738da"
                    ]
                ],
                "2": [
                    "versions.yml:md5,8965d3d0e4b22a572f64f1007a2439fd"
                ],
                "tr_fasta": [
                    [
                        {
                            "id": "test"
                        },
                        "test_trfinder.fasta.gz:md5,4404f9d64e3cbdced310646cbc2738da"
                    ]
                ],
                "tr_stats": [
                    [
                        {
                            "id": "test"
                        },
                        "test_trfinder.tsv:md5,e5bebaac45561e6a7a2a3d88baa42985"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8965d3d0e4b22a572f64f1007a2439fd"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.1"
        },
        "timestamp": "2024-07-24T08:30:12.954251239"
    }
}